sandbox/examples/demo/_grading/steno.rb

```diff
@@ -0,0 +1,105 @@
+require 'open3'
+require 'digest'
+require 'fileutils'
+
+class Steno
+  attr_accessor :sub
+
+  def initialize
+    $stdout.sync = true
+    $stderr.sync = true
+
+    @cookie  = ENV.delete("COOKIE")  || "COOKIE"
+    @sub     = ENV.delete("SUB") or raise Exception.new("Must set SUB env var")
+    @timeout = (ENV.delete("TIMEOUT") || 60).to_i
+    @ktime   = @timeout + 10
+    @hashes  = {}
+  end
+
+  def hash_file(path)
+    Digest::SHA256.digest(File.read(path))
+  end
+
+  def save_grading_hashes(xs)
+    xs.each do |name|
+      hash = hash_file("_grading/#{name}")
+      @hashes[name] = hash
+    end
+  end
+
+  def check_grading_hashes
+    @hashes.each do |name, hash0|
+      hash1 = hash_file("_grading/#{name}")
+      if hash0 != hash1
+        puts "Hash mismatch for _grading/#{name}"
+        puts "Hax!"
+        exit(1)
+      end
+    end
+  end
+
+  def shell(cmd)
+    cmd.gsub!(%q{"}, %q{\"})
+    puts cmd
+    system(%Q{timeout -k #{@ktime} #{@timeout} bash -c "#{cmd}"}) or
+      raise Exception.new("shell(#{cmd}) -> #{$?}")
+  end
+
+  def unpack
+    if @sub =~ /\.zip$/i
```

Comment:
We can pull in ArchiveUtils to the sample _grading directory.

Do you have a malicious zip archive to test with? I'd like to make it so that the sandbox catches that error - although we'll still want ArchiveUtils so we can report clean errors.